### PR TITLE
refactor(ent): extensions

### PR DESCRIPTION
**Pull request by hekike**
Turn ent templates into extensions to make them sharable.

### CODE REVIEW

File: openmeter/ent/generate.go [hunk 1]
```diff
@@ -1,3 +1,3 @@
 package entdb
 
-//go:generate go run -mod=mod entgo.io/ent/cmd/ent  generate --target ./db  --feature sql/upsert --feature sql/lock --feature sql/versioned-migration --template ../../pkg/framework/entutils/expose.tpl --template ../../pkg/framework/entutils/paginate.tpl ./schema
+//go:generate go run -mod=mod entc.go
```
Comment (chrisgacsal) [inline openmeter/ent/generate.go:3]:
```suggestion
//go:generate go run -mod=readonly entc.go
```

Comment (chrisgacsal) [reply openmeter/ent/generate.go:3]: Just to avoid automatic pkg updates during `go generate`.

**Review by chrisgacsal — APPROVED** [review]

File: pkg/framework/entutils/entexpose/entexpose.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package entexpose
```
Comment (GAlexIHU) [inline pkg/framework/entutils/entexpose/entexpose.go:1]: nit: i'd keep `entexpose` and `entpaginate` in the same package, maybe simply `extensions`

**Review by GAlexIHU — APPROVED** [review]
Neat!

File: pkg/framework/entutils/entexpose/entexpose.go (hunk 2 above)
Comment (hekike) [reply pkg/framework/entutils/entexpose/entexpose.go:1]: I prefer them separate, as nit I'll skip on it

**Pull request merged**